scaffolds/java/scripts/build

```diff
@@ -2,6 +2,6 @@
 
 cd $(dirname $0)/..
 
-./gradle build -x test
+./gradle.sh assemble # or "./gradlew assemble" if you don't want to use Docker
```

Comment:
Your comment makes me think that we could/should make `./gradle` (AKA `./gradle.sh` or `./gradle-env`) capable of switching automatically between the different options:

```bash
#!/bin/bash

cd "$(dirname "$0")" || exit 1

if [[ -f ./gradlew && -x ./gradlew  ]]; then
  ./gradlew "$@"
elif which gradle > /dev/null; then
  gradle "$@"
else
  docker run --rm \
    -u $(id -u):$(id -g) \
    -v "$(pwd)":/home/gradle/project:rw \
    -w /home/gradle/project \
    gradle:7.4.2-jdk11-alpine gradle "$@"
fi
```

WDYT ?

Reply:
You're implying that if a developer wants to use Docker, then he has no `gradlew` nor Gradle installed, and I think it's wrong.
Developers may firstly use `gradlew`, then for any reason, may want to give a try with Docker, then the `build|test` scripts will still use `gradlew`. And I don't want to tell the developer that using Docker and having `gradlew` are mutually exclusive.

Reply:
> You're implying that if a developer wants to use Docker, then he has no gradlew nor Gradle installed, and I think it's wrong.

I'm not implying anything :-). Just to clarify: the reason I suggest this script is that it would be a great way to make the TCR scripts work automatically with whatever the user has available to run the tests.

In any case, it seems highly probable to me, that someone that has installed gradle or a gradle wrapper did it with the intent to use it. The docker wrapper, in my view, is a fallback for those who do not wish to install the development environment for a given language. 

> And I don't want to tell the developer that using Docker and having gradlew are mutually exclusive.

I don't believe that's what we're saying here, and I don't believe this script makes them mutually exclusive: you're always free to run `./gradlew` or a `docker run ...` yourself. The script is not here to cover all usages (that would be unrealistic), but to cover the most common ones.

The objective of these scaffolds is to get started quickly, without any fuss. The point is to go from zero to coding very quickly, whether a dev env is present on your machine or not. Everything that is handled automatically is a plus. Every choice that must be made is friction we want to avoid.

If a dev has already taken the time to install a `./gradlew`, it's a good indicator that this is how they would like to work for this repo. If it's not any more, they can delete the wrapper or call the docker image directly.

Reply:
> Every choice that must be made is friction we want to avoid.

Ok I get your point. You have a practical point of view, while I have a more pedagogic one.

At the end, this repo should follow a single vision. I'll add you script.